packages/browser-utils/src/focus.ts

```diff
@@ -95,8 +95,12 @@ export function trapFocus(element: Element): FocusTrap {
 
   return {
     deactivate() {
-      document.removeEventListener('keydown', keyDown);
-      (previousActiveElement as HTMLElement)?.focus();
+      // Restore focus to the previously active element, but only if focus is
+      // currently inside the trap container.
+      if (element.contains(document.activeElement)) {
+        document.removeEventListener('keydown', keyDown);
```

Comment:
Do we really want to retain the event listener if this condition is false?

I was able to reproduce the weird behavior before this fix by removing the IP address link from the fingerprint popover and then opening a popover. Since there's nothing focusable in it, I was able to tab around to other popovers on the page. Opening one would then shift focus back to the last trigger, which is definitely broken. I can confirm that that works correctly after this change.

Reply:
> Do we really want to retain the event listener if this condition is false?

Good catch, this was an oversight. Fixed in 60dbe2b386eca6a6ef656c68f0135fa729d7865e.